### PR TITLE
Fix documents to reflect `CUPY_ACCELERATORS`

### LINKED ISSUE #3586: Change docs to reflect `CUPY_ACCELERATORS`
In #3461 the new env var `CUPY_ACCELERATORS` is documented. However, the purpose and usage of reduction accelerators should also be documented. Perhaps we can just update this description to make the distinction between routine and reduction accelerators?
https://github.com/cupy/cupy/blob/50bac854269742c5eed277208e7310ea38167a80/docs/source/reference/routines.rst#L33-L41

### PR DESCRIPTION
**Pull request by asi1024**
Resolve #3586.

### CODE REVIEW

File: docs/source/reference/routines.rst [hunk 1]
```diff
@@ -30,14 +30,14 @@ These functions cover a subset of
...
 :func:`~cupy.argmin`, :func:`~cupy.argmax`, and other functions built on top of them, can be
-accelerated by switching to the `CUB`_ backend. The switch can be toggled on or off at runtime
-by setting the bool :data:`cupy.cuda.cub_enabled`, which is set to ``False`` by default. Note
-that while in general CUB-backed reductions are faster, there could be exceptions depending on
-the data layout. We recommend users to perform some benchmarks to determine whether CUB offers
-better performance or not.
+accelerated by switching to the `CUB`_ or `cuTENSOR`_ backend. These backends can be enabled
+by setting ``CUPY_ACCELERATORS`` environement variable. Note that while in general the accelerated
+reductions are faster, there could be exceptions depending on the data layout. We recommend
+users to perform some benchmarks to determine whether CUB offers better performance or not.
```
Comment (leofang) [inline docs/source/reference/routines.rst:40]:
```suggestion
users to perform some benchmarks to determine whether CUB/cuTENSOR offers better performance
or not.
```

File: docs/source/reference/routines.rst [hunk 2]
```diff
@@ -30,14 +30,14 @@ These functions cover a subset of
...
 ----------------------------------
 Some CuPy reduction routines, including :func:`~cupy.sum`, :func:`~cupy.min`, :func:`~cupy.max`,
 :func:`~cupy.argmin`, :func:`~cupy.argmax`, and other functions built on top of them, can be
-accelerated by switching to the `CUB`_ backend. The switch can be toggled on or off at runtime
-by setting the bool :data:`cupy.cuda.cub_enabled`, which is set to ``False`` by default. Note
-that while in general CUB-backed reductions are faster, there could be exceptions depending on
-the data layout. We recommend users to perform some benchmarks to determine whether CUB offers
-better performance or not.
+accelerated by switching to the `CUB`_ or `cuTENSOR`_ backend. These backends can be enabled
+by setting ``CUPY_ACCELERATORS`` environement variable. Note that while in general the accelerated
```
Comment (leofang) [inline docs/source/reference/routines.rst:38]: Is it possible to add a hyperlink to ``CUPY_ACCELERATORS`` in the environment variable page?

**Comment by kmaehashi** [comment]
@asi1024 Could you reflect these too? :pray:
https://github.com/cupy/cupy/issues/3530#issuecomment-675624144
https://github.com/cupy/cupy/issues/3530#issuecomment-675625765

**Comment by asi1024** [comment]
PTAL.

**Comment by leofang** [comment]
```
Warning, treated as error:
343/home/travis/build/cupy/cupy/docs/source/reference/routines.rst:34:Title underline too short.
```

File: docs/source/reference/routines.rst [hunk 3]
```diff
@@ -30,14 +30,15 @@ These functions cover a subset of
    ext
 
 
-CUB backend for reduction routines
-----------------------------------
+CUB/cuTENSOR backend for reduction routines
+-------------------------------------------
 Some CuPy reduction routines, including :func:`~cupy.sum`, :func:`~cupy.min`, :func:`~cupy.max`,
```
Comment (leofang) [inline docs/source/reference/routines.rst:35]: Do we know why `min` and `max` are not rendered with hyperlinks in RTD?

File: docs/source/reference/environment.rst [hunk 4]
```diff
@@ -65,7 +65,9 @@ Here are the environment variables CuPy uses.
 +------------------------------------+----------------------------------------------------+
 
 Moreover, as in any CUDA programs, all of the CUDA environment variables listed in the `CUDA Toolkit
-Documentation`_ will also be honored.
+Documentation`_ will also be honored. When ``CUPY_ACCELERATORS`` or ``NVCC`` environment variables
+are set, g++-6 or later is required as the runtime host compiler. Please refer
```
Comment (leofang) [inline docs/source/reference/environment.rst:69]:
```suggestion
are set, g++-6 or later is required as the runtime host compiler. Please refer to
```

File: docs/source/reference/routines.rst (hunk 3 above)
Comment (kmaehashi) [reply docs/source/reference/routines.rst:35]: `numpy.{min,max}` are undocumented functions so cupy is not documenting them as well.

**Comment by kmaehashi** [comment]
LGTM! pfnCI, test this please.

File: docs/source/reference/routines.rst (hunk 3 above)
Comment (kmaehashi) [reply docs/source/reference/routines.rst:35]: Maybe we should point to amax/amin instead?

Comment (leofang) [reply docs/source/reference/routines.rst:35]: What! I didn't know that...Interesting...Yeah sounds good.

**Comment by asi1024** [comment]
pfnCI, test this please.

**Pull request merged**